zipline/testing/fixtures.py

```diff
@@ -315,72 +333,218 @@ def init_class_fixtures(cls):
+                    cls.BCOLZ_MINUTE_BAR_LOOKBACK_DAYS
+                ],
+                cls.sim_params.period_end,
+            )
+        )
         writer = BcolzMinuteBarWriter(
-            cls.env.first_trading_day,
-            cls.tempdir.path,
-            cls.env.open_and_closes.market_open,
-            cls.env.open_and_closes.market_close,
-            cls.BCOLZ_MINUTES_PER_DAY,
+            days[0],
+            p,
+            cls.env.open_and_closes.market_open.loc[days],
+            cls.env.open_and_closes.market_close.loc[days],
+            US_EQUITIES_MINUTES_PER_DAY
+        )
+        cls.bcolz_minute_bar_data = df_dict = cls.make_minute_bar_data()
+        for sid, df in iteritems(df_dict):
+            writer.write(sid, df)
+
+        cls.bcolz_minute_bar_reader = BcolzMinuteBarReader(p)
+
+
+class WithAdjustmentReader(WithBcolzDailyBarReader):
+    """
+    ZiplineTestCase mixin providing cls.adjustment_reader as a class level
+    fixture.
+
+    After init_class_fixtures has been called, `cls.adjustment_reader` will be
+    populated with a new SQLiteAdjustmentReader object. The data that will be
+    written can be passed by overriding `make_{field}_data` where field may
+    be `splits`, `mergers` `dividends`, or `stock_dividends`.
+    The daily bar reader used for this adjustment reader may be customized
+    by overriding `make_adjustment_writer_daily_bar_reader`. This is useful
+    to providing a `MockDailyBarReader`.
+
+    For more advanced configuration, `make_adjustment_writer` may be
+    overwritten directly.
+
+    See Also
+    --------
+    zipline.testing.MockDailyBarReader
+    """
+    @classmethod
+    def _make_data(cls):
+        return None
+
+    make_splits_data = _make_data
+    make_mergers_data = _make_data
+    make_dividends_data = _make_data
+    make_stock_dividends_data = _make_data
+
+    del _make_data
+
+    @classmethod
+    def make_adjustment_writer(cls):
+        return SQLiteAdjustmentWriter(
+            cls.adjustment_db_conn,
+            cls.make_adjustment_writer_daily_bar_reader(),
+            cls.bcolz_daily_bar_days,
         )
-        for sid, data in iteritems(cls.make_bcolz_minute_bar_data()):
-            writer.write(sid, data)
-        cls.bcolz_minute_bar_reader = BcolzMinuteBarReader(cls.tempdir.path)
 
     @classmethod
-    def make_bcolz_minute_bar_data(cls):
-        """
-        Returns
-        -------
-        A dict of sid -> DataFrame with columns ('open', 'high', 'low',
-        'close', 'volume') and an index of the minutes on which the prices
-        occurred.
+    def make_adjustment_writer_daily_bar_reader(cls):
+        return cls.bcolz_daily_bar_reader
 
-        See, zipline.data.minute_bars.BcolzMinuteBarWriter
-        """
-        return {}
+    @classmethod
+    def init_class_fixtures(cls):
+        super(WithAdjustmentReader, cls).init_class_fixtures()
+        cls.adjustment_db_conn = conn = sqlite3.connect(':memory:')
```

Comment:
Does anyone use this attribute?  Could we just make it an arg to `make_adjustment_writer`?  The only case where that wouldn't work would be if we wanted to test writing into an already-populated db, but in that case we probably shouldn't use the fixture.